app/services/user_factory.rb

```diff
@@ -0,0 +1,13 @@
+class UserFactory
+  def initialize(authenticator)
+    @authenticator = authenticator
+  end
+
+  def find_or_create_user
+    User.find_or_create_by(name: authenticator.name)
+  end
+
+  private
+
+  attr_reader :authenticator
+end
```

Comment:
I have it changed on my local copy but it's not pushing up the revised version with the newline added, not sure why.